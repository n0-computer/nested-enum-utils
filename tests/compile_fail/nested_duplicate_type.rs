use nested_enum_utils::enum_conversions;

#[derive(Debug)]
#[enum_conversions(Outer)]
enum Inner1 {
    A(u32),
    B(u8),
}

#[derive(Debug)]
#[enum_conversions(Outer)]
enum Inner2 {
    A(u32),
    B(u8),
}

#[derive(Debug)]
#[enum_conversions]
enum Outer {
    A(Inner1),
    B(Inner2),
}

fn main() {}
