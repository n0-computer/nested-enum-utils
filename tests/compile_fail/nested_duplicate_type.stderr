error[E0119]: conflicting implementations of trait `From<u32>` for type `Outer`
  --> tests/compile_fail/nested_duplicate_type.rs:11:1
   |
4  | #[enum_conversions(Outer)]
   | ------------------------ first implementation here
...
11 | #[enum_conversions(Outer)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Outer`
   |
   = note: this error originates in the attribute macro `enum_conversions` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `From<u8>` for type `Outer`
  --> tests/compile_fail/nested_duplicate_type.rs:11:1
   |
4  | #[enum_conversions(Outer)]
   | ------------------------ first implementation here
...
11 | #[enum_conversions(Outer)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Outer`
   |
   = note: this error originates in the attribute macro `enum_conversions` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `TryFrom<Outer>` for type `u32`
  --> tests/compile_fail/nested_duplicate_type.rs:11:1
   |
4  | #[enum_conversions(Outer)]
   | -------------------------- first implementation here
...
11 | #[enum_conversions(Outer)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `u32`
   |
   = note: this error originates in the attribute macro `enum_conversions` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `TryFrom<&Outer>` for type `&u32`
  --> tests/compile_fail/nested_duplicate_type.rs:11:1
   |
4  | #[enum_conversions(Outer)]
   | -------------------------- first implementation here
...
11 | #[enum_conversions(Outer)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `&u32`
   |
   = note: this error originates in the attribute macro `enum_conversions` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `TryFrom<Outer>` for type `u8`
  --> tests/compile_fail/nested_duplicate_type.rs:11:1
   |
4  | #[enum_conversions(Outer)]
   | -------------------------- first implementation here
...
11 | #[enum_conversions(Outer)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `u8`
   |
   = note: this error originates in the attribute macro `enum_conversions` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `TryFrom<&Outer>` for type `&u8`
  --> tests/compile_fail/nested_duplicate_type.rs:11:1
   |
4  | #[enum_conversions(Outer)]
   | -------------------------- first implementation here
...
11 | #[enum_conversions(Outer)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `&u8`
   |
   = note: this error originates in the attribute macro `enum_conversions` (in Nightly builds, run with -Z macro-backtrace for more info)
